application/cms/models.py

```diff
@@ -1039,26 +1056,44 @@ class Measure(db.Model):
     reference = db.Column(db.String(32), nullable=True)  # optional internal reference
 
     # relationships
-    subtopics = db.relationship("Subtopic", secondary="subtopic_measure", back_populates="measures")
+    subtopics = db.relationship(
+        "Subtopic", secondary="subtopic_measure", back_populates="measures", order_by="asc(Subtopic.id)"
+    )
+    versions = db.relationship("MeasureVersion", back_populates="measure", order_by=desc(MeasureVersion.version))
 
     # Departmental users can only access measures that have been shared with them, as defined by this relationship
-    # TODO: Uncomment this and use back_populates (relationship declared both sides) once user_measure table exists
-    # shared_with = db.relationship(
-    #     "User",
-    #     lazy="subquery",
-    #     secondary=user_measure,
-    #     primaryjoin="Measure.id == user_measure.columns.measure_id",
-    #     secondaryjoin="User.id == user_measure.columns.user_id",
-    #     backref=db.backref("measures", lazy=True),
-    # )
-
-    # TODO: Uncomment these once MeasureVersion exists
-    # def get_versions(self):
-    #     return (
-    #         MeasureVersion.query.filter(MeasureVersion.measure_id == self.id)
-    #         .order_by(desc(MeasureVersion.version))
-    #         .all()
-    #     )
-    #
-    # def latest_published_version_id(self):
-    #     return self.get_versions()[0].id
+    shared_with = db.relationship(
+        "User",
+        lazy="subquery",
+        secondary="user_measure",
+        primaryjoin="Measure.id == user_measure.columns.measure_id",
+        secondaryjoin="User.id == user_measure.columns.user_id",
+        back_populates="measures",
+    )
+
+    @property
+    def has_published_version(self):
+        return any(mv.published for mv in self.versions)
+
+    @property
+    def latest_version(self):
+        return MeasureVersion.query.filter(MeasureVersion.measure_id == self.id, MeasureVersion.latest == True).one()
+
+    @property
+    def latest_published_version(self):
+        published_versions = MeasureVersion.query.filter(
+            MeasureVersion.measure_id == self.id, MeasureVersion.published == True
+        ).all()
+
+        if published_versions:
+            return max(published_versions, key=lambda x: x.version)
+
+        return None
+
+    @property
+    def subtopic(self):
+        return self.subtopics[0]
+
+    @property
+    def title(self):
```

Comment:
This is something Kev and I threw in as a shortcut. Maybe it needs to be removed so that we need to explicitly reference a version all the time (e.g. latest draft, or latest published). That would probably be clearer and more sensible in the long run?

Reply:
I've got rid of this now - it was confusing me. So you now have to do `measure.latest_version.title` explicitly. It feels clearer.